### PR TITLE
Bugfix/heap corruptions

### PR DESCRIPTION
**Pull request by victorlevasseur**
Fix some crashes related to out of range access or after destruction access.

### CODE REVIEW

File: Core/GDCore/Events/Builtin/LinkEvent.cpp [hunk 1]
```diff
@@ -44,6 +44,7 @@ const EventsList * LinkEvent::GetLinkedEvents(const gd::Project & project) const
 
 void LinkEvent::ReplaceLinkByLinkedEvents(gd::Project & project, EventsList & eventList, std::size_t indexOfTheEventInThisList)
 {
+    linkWasInvalid = false;
```
Comment (4ian) [inline Core/GDCore/Events/Builtin/LinkEvent.cpp]: @victorlevasseur Any reason for moving this line? :)

File: Core/GDCore/IDE/wxTools/FlatAuiTabArt.cpp [hunk 2]
```diff
@@ -578,8 +578,8 @@ void FlatAuiTabArt::DrawTab(wxDC& dc,
 
         dc.DrawLine(border_points[0].x+1,
                     border_points[0].y,
-                    border_points[5].x,
-                    border_points[5].y);
+                    border_points[3].x,
+                    border_points[3].y);
```
Comment (4ian) [inline Core/GDCore/IDE/wxTools/FlatAuiTabArt.cpp]: oO Good catch!

File: Core/GDCore/Events/Builtin/LinkEvent.cpp (hunk 1 above)
Comment (victorlevasseur) [inline Core/GDCore/Events/Builtin/LinkEvent.cpp]: Not to do it after the event is destroyed by itself. See line 82/83.

File: Core/GDCore/IDE/wxTools/FlatAuiTabArt.cpp (hunk 2 above)
Comment (victorlevasseur) [inline Core/GDCore/IDE/wxTools/FlatAuiTabArt.cpp]: gdb emitted a SIGTRAP on Windows (only ?!) on this line.

File: Core/GDCore/Events/Builtin/LinkEvent.cpp (hunk 1 above)
Comment (4ian) [inline Core/GDCore/Events/Builtin/LinkEvent.cpp]: Ok very true! Maybe a comment should be added after these lines, or at the end of the function to tell that the event was potentially just destroyed?

**Comment by 4ian** [comment]
Looks good for merging otherwise! Good job :)

**Comment by victorlevasseur** [comment]
Not really related but, any reason the ```EventList``` is a list of ```std::shared_ptr``` and not ```std::unique_ptr``` ?

**Comment by 4ian** [comment]
@victorlevasseur No particular reason, just that I'm used to std::shared_ptr :)

**Pull request merged**